tests/Integration/Behaviour/Features/Context/Domain/MetaFeatureContext.php

```diff
@@ -0,0 +1,287 @@
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace Tests\Integration\Behaviour\Features\Context\Domain;
+
+use Behat\Gherkin\Node\TableNode;
+use Exception;
+use Meta;
+use PrestaShop\PrestaShop\Core\Domain\Meta\Command\AddMetaCommand;
+use PrestaShop\PrestaShop\Core\Domain\Meta\Command\EditMetaCommand;
+use PrestaShop\PrestaShop\Core\Domain\Meta\Exception\MetaConstraintException;
+use PrestaShop\PrestaShop\Core\Domain\Meta\Exception\MetaNotFoundException;
+use PrestaShop\PrestaShop\Core\Domain\Meta\Query\GetMetaForEditing;
+use PrestaShop\PrestaShop\Core\Domain\Meta\Query\GetPagesForLayoutCustomization;
+use PrestaShop\PrestaShop\Core\Domain\Meta\QueryResult\EditableMeta;
+use PrestaShop\PrestaShop\Core\Domain\Meta\QueryResult\LayoutCustomizationPage;
+use PrestaShop\PrestaShop\Core\Domain\Meta\ValueObject\MetaId;
+use RuntimeException;
+use Tests\Integration\Behaviour\Features\Context\SharedStorage;
+use Tests\Integration\Behaviour\Features\Context\Util\DataTransfer;
+
+class MetaFeatureContext extends AbstractDomainFeatureContext
+{
+    /**
+     * @When /^I add meta "([^"]*)" with specified properties$/
+     */
+    public function addMetaWithSpecifiedProperties($reference, TableNode $node)
+    {
+        $data = $node->getRowsHash();
+        $data = $this->getWithDefaultLanguage($data);
+
+        $command = new AddMetaCommand($data['pageName']);
+        $command = DataTransfer::transferAttributesFromArrayToObject($data, $command);
+
+        try {
+            /** @var MetaId $metaId */
+            $metaId = $this->getCommandBus()->handle($command);
+
+            SharedStorage::getStorage()->set($reference, new Meta($metaId->getValue()));
+        } catch (Exception $exception) {
+            $this->lastException = $exception;
+            $this->lastErrorCode = $exception->getCode();
+        }
+    }
+
+    /**
+     * @When /^I get meta "([^"]*)" with specified properties$/
+     */
+    public function getMetaWithSpecifiedProperties($reference, TableNode $node)
+    {
+        $data = $node->getRowsHash();
+        $queryCommand = new GetMetaForEditing((int) $data['metaId']);
+
+        try {
+            /** @var EditableMeta $editableMeta */
+            $editableMeta = $this->getQueryBus()->handle($queryCommand);
+
+            SharedStorage::getStorage()->set("editable_{$reference}", $editableMeta);
```

Comment:
Why prefix with `editable` ? Why not using `$reference` directly ?

Reply:
in behat scenarios I use same references "meta1" , "meta2" and "meta3" which actually represent data which is stored in objectModel "Meta". Hovewer this reference is retrieved from Query `GetMetaForEditing` and it is stored as object `EditableMeta` so I prefix it differently because it holds kind of different object.